core/slice.go

```diff
@@ -751,7 +751,7 @@ func (sl *Slice) ConstructLocalMinedBlock(header *types.Header) (*types.Block, e
 		// in the subordinate manifest.
 		return types.NewBlockWithHeader(header), nil
 	}
-	pendingBlockBody := sl.GetPendingBlockBody(header.Root())
+	pendingBlockBody := sl.GetPendingBlockBody(sl.miner.worker.PendingBlockBodyKey(header))
```

Comment:
I would prefer to pass in a header here, and let `GetPendingBlockBody()` construct the key from the header, but that's just a nit-pick. I'm fine with this as is if you disagree.

Reply:
we do pass in header. but the method is in worker. i can create a wrapper function in slice if that makes it look neat

Reply:
Updated all the helper functions. Please take a look again.